packages/ui/react-ui-editor/src/components/TextEditor/hooks.stories.tsx

```diff
@@ -74,11 +76,13 @@ const Story = ({ autoFocus, placeholder, doc, readonly }: StoryProps) => {
 
   const handleAction = useActionHandler(view);
 
+  // FIXME This doesn't update the state on view changes. Also not
+  // sure if view is even guaranteed to exist at this point.
   return (
     <div className={mx(fixedInsetFlexLayout, groupSurface)}>
       <div className='flex h-full justify-center'>
         <div className='flex flex-col h-full w-[800px]'>
-          <Toolbar.Root onAction={handleAction}>
+          <Toolbar.Root onAction={handleAction} state={view ? getFormatting(view.state) : emptyFormatting}>
```

Comment:
`view && getFormatting(view.state)`
undefined should be handled by caller which then decides on the appropriate no-op.
e.g.,

```
getState={(s) => s?.strikethrough ?? false}
```

i think the modern `?` and `??` operators make it clearer how to handle undefined as an expected state.